Application start fails with NullPointerException because of MicrometerRecorder changes (3.3.0)

### Describe the bug

With Quarkus 3.3.0 our application cannot start anymore and fails with a NullPointerException:

```java
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.size()" because "classSpecificCustomizers" is null
	at io.quarkus.micrometer.runtime.MicrometerRecorder.configureRegistries(MicrometerRecorder.java:99)
	at io.quarkus.deployment.steps.MicrometerProcessor$configureRegistry477897979.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.MicrometerProcessor$configureRegistry477897979.deploy(Unknown Source)
```

We have a custom MeterRegistry (DynatraceMeterRegistry). We configured it exactly like this:
https://quarkus.io/guides/telemetry-micrometer#create-a-customized-meterregistry

After debugging I could see that ```registryClasses```
https://github.com/quarkusio/quarkus/blob/2c6f8b84a115fbab00e562591a277dbeb8869c8d/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerRecorder.java#L65

Only contains two registries:
- PrometheusMeterRegistry
- JsonMeterRegistry

But in this line:
https://github.com/quarkusio/quarkus/blob/2c6f8b84a115fbab00e562591a277dbeb8869c8d/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerRecorder.java#L80

Quarkus finds three beans:
1. PrometheusMeterRegistryProducer...
2. JsonMeterRegistryProvider...
3. DynatraceMicrometerRegistry...

The loop over the beans with three entries then tries to find the ```classSpecificCustomizers``` for ```DynatraceMeterRegistry```:
https://github.com/quarkusio/quarkus/blob/2c6f8b84a115fbab00e562591a277dbeb8869c8d/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerRecorder.java#L97

This returns ```null```, which results in a ```NullPointerException``` in line:
https://github.com/quarkusio/quarkus/blob/2c6f8b84a115fbab00e562591a277dbeb8869c8d/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerRecorder.java#L99


### Expected behavior

Our application can start as usual 😉 .

### Actual behavior

See bug description.

### How to Reproduce?

See bug description.

### Output of `uname -a` or `ver`

Darwin <PC> 22.5.0 Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk 17.0.7 2023-04-18 LTS

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.3.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.9.2

### Additional information

_No response_

--- a/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerRecorder.java
+++ b/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerRecorder.java
@@ -94,7 +94,8 @@ public class MicrometerRecorder {
                 Class<?> registryClass = registry.getClass();
                 applyMeterFilters(registry, classMeterFilters.get(registryClass));
 
-                var classSpecificCustomizers = classMeterRegistryCustomizers.get(registryClass);
+                var classSpecificCustomizers = classMeterRegistryCustomizers.getOrDefault(registryClass,
+                        Collections.emptyList());
                 var newList = new ArrayList<MeterRegistryCustomizer>(
                         globalMeterRegistryCustomizers.size() + classSpecificCustomizers.size());
                 newList.addAll(globalMeterRegistryCustomizers);

